src/GeoExt/data/proxy/Protocol.js

```diff
@@ -63,22 +103,28 @@ Ext.define('GeoExt.data.proxy.Protocol', {
      */
     doRequest: function(operation, callback, scope) {
         var me = this,
+            operationParams,
             params,
-            request;
+            request,
+            o,
+            cb,
+            options;
 
         if(GeoExt.isExt5){
-            callback = operation.getCallback();
-            scope = operation.getScope();
-            params = Ext.applyIf(operation.getParams() || {}, me.extraParams || {});
-        }
-        else {
-            params = Ext.applyIf(operation.params || {}, me.extraParams || {});
+            callback = callback || operation.getCallback();
+            scope = scope || operation.getScope();
```

Comment:
Good question. I am not 100% sure to be honest. In my current understanding I'd say that we should have had something similar in ExtJS 4 all the time. This is certainly a thing we should look at once we start rereviewing the branch in total.